server/plugins/FilterUGens.cpp

```diff
@@ -1053,7 +1053,15 @@ void Integrator_Ctor(Integrator* unit) {
         SETCALC(Integrator_next);
     unit->m_b1 = ZIN0(1);
     unit->m_y1 = 0.f;
-    Integrator_next(unit, 1);
+    // Note: Here, we deliberately do NOT want to call Integrator_next(unit, 1).
+    // That would initialize the presample AND update m_y1.
+    // Then this initial value will be added in *again* by the normal calc loop.
+    // So the integral would be corrupted for its lifetime.
+    // The formula is input + (coef * output[-1]).
+    // But output[-1] is 0 initially, so we just drop that term.
+    float* out = ZOUT(0);
+    float* in = ZIN(0);
+    ZXP(out) = ZXP(in);
```

Comment:
I just noticed: no need for advancing pointers when we're just getting/setting one sample:
```
ZOUT(0) = ZIN(0); // out = in + (coef * out[-1]); 
```
> Exceptional cases probably should be noted with a comment.

I agree. In the case of setting the output sample directly, I think it could be a good idea to note what the calculation "formula" is in a comment, in a way indicating that the init sample == first sample concern  has been considered. (I added this to the code above as an example.)

What do you think? Just thinking out loud because this discussion will inform how I continue to handle this type of bug. Perhaps I'll move this over to the [RFC](https://github.com/supercollider/rfcs/pull/19) for coming up with some standard approaches to this.